mmv1/products/bigqueryconnection/Connection.yaml

```diff
@@ -299,10 +310,43 @@ properties:
       - azure
       - cloud_spanner
       - cloud_resource
+      - spark
+    send_empty_value: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'serviceAccountId'
+        description: The account ID of the service created for the purpose of this
+          connection.
+        output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: spark
+    description: Container for connection properties to execute stored procedures for Apache Spark.
+      resources.
+    exactly_one_of:
+      - cloud_sql
+      - aws
+      - azure
+      - cloud_spanner
+      - cloud_resource
+      - spark
     send_empty_value: true
     properties:
       - !ruby/object:Api::Type::String
         name: 'serviceAccountId'
         description: The account ID of the service created for the purpose of this
           connection.
         output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: metastoreServiceConfig
+        description: Dataproc Metastore Service configuration for the connection.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: metastoreService
+            description: Resource name of an existing Dataproc Metastore service in the form of projects/[projectId]/locations/[region]/services/[serviceId].
+      - !ruby/object:Api::Type::NestedObject
+        name: sparkHistoryServerConfig
+        description: Spark History Server configuration for the connection.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: dataprocCluster
+            description: Resource name of an existing Dataproc Cluster to act as a Spark History Server for the connection if the form of projects/[projectId]/regions/[region]/clusters/[cluster_name].
```

Comment:
These values are settable but there is no corresponding test. Could we test these sub-properties in a configuration?

Reply:
Sure, it will make the test at least 20 mins though. ill add those

Reply:
mm I see. As we need to spin up a dataproc cluster for them.

Reply:
added dataproc test, i can add metastore as well but its going to take a long time to test. wdyt @ScottSuarez ?

Reply:
let's abstain for now ! As long as you've verified locally